runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -324,7 +325,8 @@ public void syncObjectUnsafe(long timestamp) {
                     try {
                         rollbackObjectUnsafe(timestamp);
                     } catch (NoRollbackException nre) {
-                        log.warn("SyncObjectUnsafe[{}] to {} failed {}", this, timestamp, nre);
+                        log.warn("SyncObjectUnsafe[{}] to {} failed {}, cause {}", this, timestamp, nre,
+                                AddressSpaceView.removalCauseCache.getIfPresent(nre.getUndoableEntryVersion()));
```

Comment:
Accessing a static variable inside the AddressSpaceView from this context seems a bit sloppy, but it is necessary (to provide important logging information) with the current code structure. Let me know if you have better ways of achieving this.